ipn/ipnlocal/local.go

```diff
@@ -463,10 +464,18 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
 		return
 	}
 
+	keyExpired := st.NetMap != nil && !st.NetMap.Expiry.IsZero() && time.Now().After(st.NetMap.Expiry)
 	b.mu.Lock()
 	wasBlocked := b.blocked
+	wasKeyExpired := b.keyExpired
+	b.keyExpired = keyExpired
```

Comment:
I think you only want to mutate b.keyExpired if st.NetMap != nil.  Otherwise it should be left how it was.

Reply:
good point. makes the logic easier to follow too.